File: 백준/Silver/5525. IOIOI/IOIOI.py
"""
[백준]
5525. IOIOI
https://www.acmicpc.net/problem/5525
"""

import heapq
import sys
from collections import deque
# 무한대를 의미하는 값으로 충분히 큰 수를 설정합니다.
#INF = float('INF')

#input = sys.stdin.readline
#print = sys.stdout.write

n = int(input())    # "IO" 패턴의 개수
m = int(input())    # 문자열의 길이
#li = list(map(int, input().split()))
s = input()         # 주어진 문자열

ans = 0
cnt = 0
i = 0

# 슬라이딩 윈도우를 사용하여 찾기
while i < m-1:
    if s[i:i+3] == "IOI":
        cnt += 1
        if cnt == n:
            ans+=1
            cnt -= 1
        i+=2

    else:
        cnt = 0
        i+=1

print(ans)

"""기존 방식: 문자열 그대로 찾기"""
"""
IOI = ""

for _ in range(n):
    IOI+="IO"
IOI+="I"
#print(IOI)

for i in range(m-len(IOI)+1):
    if s[i]==s[i+1]:
        continue
    if s[i:i+len(IOI)] == IOI:
        ans+=1
print(ans)
"""

File: 백준/Silver/5525. IOIOI/IOIOI.cc
#include <iostream>

using namespace std;

int main() {
    // N+1개의 I와 N개의 O
    // n: 1 => I 2개와 O 1개인 IOI
    // n: 2 => I 3개와 O 2개인 IOIOI

    int n, len;
    cin >> n >> len;

    string str;
    cin >> str;

    // 찾아야 하는 문자열 만들기
    string find = "";
    int findLen = (n + 1) + n;

    while (findLen--) {
        if (findLen % 2 == 0)   // 짝수번째는 I
            find += "I";
        else
            find += "O";
    }

    int cnt = 0;
    findLen = (n + 1) + n;
    for (int i = 0; i <= len - findLen; i++) {
        string tmp = str.substr(i, findLen);    // i번째부터, findLen만큼 찾기

        if (tmp == find)
            cnt++;
    }
    cout << cnt << endl;
}